compiler/rustc_abi/src/lib.rs

```diff
@@ -101,6 +101,10 @@ pub struct ReprOptions {
 }
 
 impl ReprOptions {
+    pub fn rust(&self) -> bool {
+        !(self.simd() || self.c() || self.packed() || self.transparent() || self.linear())
```

Comment:
This could be a better name. `repr(packed)` is still a flavor of Rust layout.

Reply:
Do you have any suggestions? I'm struggling to come up with one, `default` doesn't seem quite right either...

Reply:
I've localized this check to `rustc_ty_utils` alongside the `noundef` eligiblity check because it's not used anywhere else and `rust()` on `ReprOptions` is as you pointed out misleading. I renamed this condition to `eligible_repr`.